File: Anexo [6]/WebViewController.m
//
//  WebViewController.m
//  TFM - Protección de Aplicaciones Móviles en iOS
//
//  Created by Felix Garcia Lainez on 18/3/18.
//  Copyright (c) 2018. All rights reserved.
//

#import <WebKit/WebKit.h>

@interface WebViewController() <WKNavigationDelegate, WKUIDelegate>

@property(nonatomic, strong) WKWebView *webView;

@end

@implementation WebViewController

- (void)viewDidLoad {
    WKPreferences* preferences = [[WKPreferences alloc] init];
    
    // Disable JavasSript execution
    [preferences setJavaScriptEnabled:NO];
    [preferences setJavaScriptCanOpenWindowsAutomatically:NO];
    
    WKWebViewConfiguration* configuration = [[WKWebViewConfiguration alloc] init];
    [configuration setPreferences:preferences];

    _webView = [[WKWebView alloc]initWithFrame:self.view.bounds configuration:configuration] ;
    [self.view addSubview:_webView];
}

@end

File: Anexo [6]/WebViewController.swift
//
//  WebViewController.swift
//  TFM - Protección de Aplicaciones Móviles en iOS
//
//  Created by Felix Garcia Lainez on 19/3/18.
//  Copyright (c) 2018. All rights reserved.
//

import UIKit
import WebKit

class WebViewController: UIViewController {
    
    private var webView: WKWebView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Disable JavasSript execution
        let preferences = WKPreferences()
        preferences.setJavaScriptEnabled(false)
        preferences.setJavaScriptCanOpenWindowsAutomatically(false)
        
        let configuration = WKWebViewConfiguration()
        configuration.preferences = preferences
        
        webView = WKWebView(frame: view.bounds, configuration: configuration)
        view.addSubview(webView)
    }
}
